File: typescript/packages/management/src/tools/get_run_trace.ts
import type { ToolContext, ToolHandler } from '../registrar.js';

export const getRunTrace: ToolHandler = async (
  ctx: ToolContext,
  input: Record<string, unknown>,
) => {
  const runId = String(input.run_id);
  return ctx.client.agents.getRun(ctx.sandboxId, runId);
};

File: python/copass-management/src/copass_management/tools/get_run_trace.py
from __future__ import annotations

from typing import Any, Dict, TYPE_CHECKING

if TYPE_CHECKING:
    from copass_management.registrar import ToolContext


async def get_run_trace(ctx: "ToolContext", input: Dict[str, Any]) -> Any:
    return await ctx.client.agents.get_run(ctx.sandbox_id, str(input["run_id"]))
